test/http-request.js

```diff
@@ -529,4 +542,25 @@ describe('httpRequest', () => {
         expect(error.message).to.be.eql(testError.message);
         expect(error instanceof MyError).to.be.eql(true);
     });
+
+    it('can read a large response using stream API', async () => {
+        const options = {
+            url: `http://${HOST}:${port}/bigFile`,
+            stream: true,
+        };
+        const response = await httpRequest(options);
+
+        const body = await readStreamToString(response);
+        expect(body).to.exist; // eslint-disable-line
```

Comment:
I think that ideally this should all happen in memory, there's really no need to create a file. Also, we should check that the source and response are actually the same, not just that the body exists.

Ideally the test would:
- create a readable stream that generates some string
- client will send this as request body to server
- server will just pipe it back
- client will download it and check that it matches what has been sent completely